aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotKeystoreHelper.java

```diff
@@ -291,13 +295,11 @@ private static KeyStore getTempKeystore(KeyStore customerKeystore, String certId
             KeyStore tempKeystore = KeyStore.getInstance(KeyStore.getDefaultType());
             tempKeystore.load(null);
 
-            X509Certificate cert = (X509Certificate) customerKeystore.getCertificate(certId);
-            tempKeystore.setCertificateEntry("cert-alias", cert);
+            X509Certificate[] certs = (X509Certificate[]) customerKeystore.getCertificateChain(certId);
```

Comment:
This is an illegal cast. `Keystore#getCertificateChain(String)` returns `Certificate[]` object, which is an array of superclass of `X509Certificate`, so casting the entire array will cause `ClassCastException` to be thrown. You would likely need to iterate through the list to cast individual item or use `Arrays.copyOf<T, U>(T[], int, Class<? extends U[]>)`.

Reply:
Hi @raphkim, I fixed the illegal cast.

`Certificate[] certs = customerKeystore.getCertificateChain(certId);
            X509Certificate[] xcerts = new X509Certificate[certs.length];

            for (int i = 0; i < certs.length; i++) {
                xcerts[i] = (X509Certificate) certs[i];
            }`